src/components/submission/list.vue

```diff
@@ -276,6 +276,13 @@ export default {
       })
         .finally(() => { this.refreshing = false; })
         .catch(noop);
+
+      // refresh keys when refreshing submissions (keys initially loaded in form/submissions)
+      if (refresh) {
```

Comment:
If we already have a key, then I think we can skip the request. Currently, it's only possible for there to be a single key, because an encrypted project can't be decrypted: `this.keys` is either an empty array or an array with one element.

```js
if (refresh && this.formVersion.keyId != null && this.keys.length === 0) {
```

Reply:
I was thinking more about this, and I think this comment wasn't entirely accurate. `keys` can only have one managed key, but I believe it could also have non-managed keys if form encryption was used before the project was encrypted. You can see in `SubmissionDownload` that `keys` is filtered for the managed key.

That said, I don't think anything here needs to change: form encryption should be rare outside of project encryption, and it seems like an edge case that a user would move from form-level encryption to project-level encryption (while also meeting other conditions needed for this whole issue to appear). The worst case is that `keys` isn't re-fetched. I wanted to make a note of this possibility though.